Docs - Create #1625 - Android app support (#3267)
microsoft/winget-cli: doc/specs/#1625 - Android app support.md

---
author: @upintheairsheep
created on: 2023-05-23
last updated: 2023-05-23
issue id: 1625
---

# Spec Title

For [#1625](https://github.com/microsoft/winget-cli/issues/1625)

## Abstract

This feature adds Android application support to winget via Windows Subsystem for Android. The goal is to allow users to install and maintain Android applications just like they would any other type of application on winget, and for more community-driven Android apps to be available via Windows.

## Inspiration

Android is by far the most used operating system on the planet, excluding a few certain regions, where iOS dominates. Many mobile developers make apps for Android, and only have a sometimes inferior web app for desktop, if they have one at all.

## Solution Design

### System requirements

Windows 11 or newer  
Windows Subsystem for Android

> ADB (if it ends up being the method of installation)

Reviewer: ```suggestion
ADB (Android Debug Bridge) (if it ends up being the method of installation)
```

> ### Android YAML manifest

Reviewer: ```suggestion
### Android Installer Type manifest
```

The "InstallerType" field in the YAML manifest file specifies the type of the application. A new InstallerType "Android" or "APK" will be added to show that a given application is an Android app.

> Each PWA would have its own manifest YAML file. A sample manifest file would look like this:

Reviewer: ```suggestion
The "InstallerType" enumeration would include "APK" to indicate an Andoroid package. A sample installer manifest file would look like the following:
```

```
Id: FDroid.FDroid
Version: 1337.0
Name: F-Droid
Publisher: F-Droid
AppMoniker: fdroid
License: Test
InstallerType: Android
Installers:
- Arch: neutral
  Url: https://f-droid.org/F-Droid.apk
  InstallerType: Android
  MinAndroidPlat: 31
  ADBFlags: 
ManifestVersion: 0.1.0
```

### Package generation

WinGetCreate will download the package and extract metadata from AndroidManifest.xml, automatically filling out all data excluding the short desciption

### Package installation

Either ADB could be used to install the application, or the method that Microsoft Store installs Android apps could be utilized.
### Flow of the install process

1. Windows Subsystem for Android is installed
2. ADB must be installed on the machine
3. It must connect to the Subsystem, bypassing the developer mode restriction, which I think the uninstallation in control panel already has.
A nice feature in the manifest would be for adding the adb install [flags](https://adbshell.com/commands/adb-install) for installation of apps that require it.
## UI/UX Design

> Installing a PWA will be similar to installing other application types currently supported by winget, using the install command. The -i flag will not be implemented due to it's impracticality.

Reviewer: ```suggestion
```
